src/ccache.cpp

```diff
@@ -2390,6 +2390,18 @@ do_cache_compilation(Context& ctx)
   // be disabled.
   Util::setenv("CCACHE_DISABLE", "1");
 
+  // If on windows check CL and _CL_ env variables and add them to
+  // the argument list
+  if (ctx.config.is_compiler_group_msvc()) {
+    if (getenv("CL")) {
+      ctx.prepend_args = Args::from_string(getenv("CL"));
+    }
+
+    if (getenv("_CL_")) {
+      ctx.append_args = Args::from_string(getenv("_CL_"));
+    }
```

Comment:
`Args::from_string` just splits on whitespace, which won't work well here since the variables are interpreted differently by the compiler. For example, if you do `set CL=/I"foo bar"` then the value needs to be parsed to `{"/Ifoo bar"}`, not `{"/I\"foo", "bar\""}`.

So part of the task is to find out how the compiler interprets the variables and implement corresponding parse functionality. It might be the same syntax as parsed by `Args::from_atfile(..., Args::AtFileFormat::msvc`, but that needs to be investigated.

Reply:
I think I'll need to rethink how to get this to work correctly,